safekeeper/src/http/routes.rs

```diff
@@ -574,7 +576,7 @@ pub fn make_router(conf: SafeKeeperConf) -> RouterBuilder<hyper::Body, ApiError>
         router = router.middleware(auth_middleware(|request| {
             #[allow(clippy::mutable_key_type)]
             static ALLOWLIST_ROUTES: Lazy<HashSet<Uri>> = Lazy::new(|| {
-                ["/v1/status", "/metrics"]
+                ["/v1/status", "/metrics", "/pprof/profile"]
```

Comment:
I didn't put this behind auth, since it's pretty inconvenient. The perf impact is minor, and only one profile can run at a time, so it's not a DDoS vector. It only exposes function and module names, so it's not a data leak risk.

Reply:
We should probably give this endpoint as well as the metrics endpoint actual JWT authentication eventually. I've filed a private issue https://github.com/neondatabase/cloud/issues/20372 as it is relevant to security.

For now I agree with you though.